File: level1/실패율.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(int N, vector<int> stages) {
    vector<int> answer;
    double count = stages.size();
    vector<pair<double, int>> result;
    double stage[N+2];

    for(int i=0; i<N+2; i++)
        stage[i] = 0;

    while(!stages.empty()){
        stage[stages.back()]++;
        stages.pop_back();
    }

    double temp;
    for(int i=1; i<N+1; i++){
        temp = stage[i];
        stage[i] /= (double)count;
        count -= temp;
    }

    for(int i=1; i<N+1; i++)
        result.push_back(make_pair((double)-1*stage[i], i));

    sort(result.begin(), result.end());

    for(int i=0; i<result.size(); i++)
        answer.push_back(result[i].second);

    return answer;
}

File: JavaScript/level1/실패율.js
function numberSort(a, b){
    if(a[0] === b[0])
        return a[1]-b[1];
     return a[0]-b[0];
}

function solution(N, stages) {
    var answer = [];
    let onStage = Array(N+2).fill(0);

    for(let n of stages)
        onStage[n]++;
    
    let triedStage = onStage.slice();
    
    for(let i=triedStage.length-1; i>0; i--)
        triedStage[i-1] += triedStage[i];

    let result = [];
    
    for(let i=1; i<onStage.length-1; i++)
        result.push([-1.0*(onStage[i]/triedStage[i]), i]);
    
    result.sort(numberSort);

    for(let s of result)
        answer.push(s[1]);
    
    return answer;
}